pkg/datatransfers/data_transfers.go

```diff
@@ -0,0 +1,197 @@
+package datatransfers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/jackc/pgx/v5"
+	_ "github.com/lib/pq"
+	"github.com/pg-sharding/spqr/pkg/config"
+	"github.com/pg-sharding/spqr/pkg/models/kr"
+	"github.com/pg-sharding/spqr/pkg/models/shrule"
+	"github.com/pg-sharding/spqr/pkg/spqrlog"
+	"github.com/pg-sharding/spqr/qdb"
+)
+
+type MoveTableRes struct {
+	TableSchema string `db:"table_schema"`
+	TableName   string `db:"table_name"`
+}
+
+// TODO: use schema
+// var schema = flag.String("shema", "", "")
+
+type ProxyW struct {
+	w io.WriteCloser
+}
+
+func (p *ProxyW) Write(bt []byte) (int, error) {
+	return p.w.Write(bt)
+}
+
+var shards *config.DatatransferConnections
+var txFrom pgx.Tx
+var txTo pgx.Tx
+var localConfigDir = "/pkg/datatransfers/shard_data.yaml"
+
+func createConnString(shardID string) string {
+	sd := shards.ShardsData[shardID]
+	return fmt.Sprintf("user=%s host=%s port=%s dbname=%s password=%s", sd.User, sd.Host, sd.Port, sd.DB, sd.Password)
+}
+
+func LoadConfig(path string) error {
+	var err error
+	shards, err = config.LoadShardDataCfg(path)
+	if err != nil {
+		p, _ := os.Getwd()
+		shards, err = config.LoadShardDataCfg(p + localConfigDir)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func MoveKeys(ctx context.Context, fromId, toId string, keyr qdb.KeyRange, shr []*shrule.ShardingRule) error {
+	if shards == nil {
+		err := LoadConfig(config.CoordinatorConfig().ShardDataCfg)
+		if err != nil {
+			return err
+		}
+	}
+
+	err := beginTransactions(ctx, fromId, toId)
+	if err != nil {
+		return err
+	}
+	defer func(ctx context.Context) {
+		err := rollbackTransactions(ctx)
+		if err != nil {
+			spqrlog.Zero.Warn().Msg("error closing transaction")
+		}
+	}(ctx)
+
+	for _, r := range shr {
+		err = moveData(ctx, *kr.KeyRangeFromDB(&keyr), r)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = commitTransactions(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func beginTransactions(ctx context.Context, f, t string) error {
+	from, err := pgx.Connect(ctx, createConnString(f))
```

Comment:
We need to create 2-phase transaction for move operation. Also, we need to reflect `Move` opertations in `QDB` to continue move operation after fail.